src/Carbon/Laravel/ServiceProvider.php

```diff
@@ -70,6 +85,34 @@ public function register()
         // Needed for Laravel < 5.3 compatibility
     }
 
+    protected function getLocale()
+    {
+        if ($this->localeGetter) {
+            return ($this->localeGetter)();
+        }
+
+        $app = $this->getApp();
+        $app = $app && method_exists($app, 'getLocale') ? $app : app('translator');
+
+        return $app->getLocale();
+    }
+
+    protected function getApp()
+    {
+        if ($this->appGetter) {
+            return ($this->appGetter)();
+        }
+
+        return \function_exists('config') && \is_array(config('octane'))
```

Comment:
Hi @nunomaduro this condition aims to target Octane installations and then in this case `app()` has the priority. Other apps keep `$this->app` as the preferred reference. Does this test would work or is there or more accurate way to know if current app is running on Octane?

Reply:
I think you can remove any octane related code here. And I will override the "appGetter" once this is released on Octane's service provider. What do you think?

Reply:
Yup, in parallel I was trying to make [CurrentApplication::set](https://github.com/kylekatarnls/octane/blob/1.x/src/CurrentApplication.php#L25) somehow change the `->app` for each of the providers when it clones the app. It would require some reflection (or some little changes in Laravel `App` class) but IMO it would be the best to solve it for all providers, not just Carbon.

Anyway we can start with appGetter and no config check, it would be indeed cleaner.

I just wonder if `Carbon\Laravel\ServiceProvider` is actually exposed in Laravel in a way that would allow you to find the right instance and call `setAppGetter()` on it?